### PR TITLE
feat: cors proxy

### PR DESCRIPTION
**Pull request by istaiti**

### CODE REVIEW

**Comment by istaiti** [comment]
Current dependencies on/for this PR:
* master
  * **PR #21** <a href="https://app.graphite.dev/github/pr/inscreen/html-to-image/21" target="_blank">Graphite</a> 
    * **PR #22** <a href="https://app.graphite.dev/github/pr/inscreen/html-to-image/22" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/inscreen/html-to-image/22?utm_source=stack-comment).

File: src/dataurl.ts [hunk 1]
```diff
@@ -74,6 +74,11 @@ export async function resourceToDataURL(
     return cache[cacheKey]
   }
 
+  if (options.useCorsProxy) {
+    // eslint-disable-next-line no-param-reassign
+    resourceUrl = options.customCorsProxyUrl + resourceUrl
```
Comment (ramonsnir) [inline src/dataurl.ts]: encode URI component?

**Review by ramonsnir — APPROVED** [review]

**Comment by istaiti** [comment]
@istaiti started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/inscreen/html-to-image/22).

**Comment by istaiti** [comment]
[Graphite](https://app.graphite.dev/github/pr/inscreen/html-to-image/22) rebased this pull request as part of a merge.

**Comment by istaiti** [comment]
@istaiti merged this pull request with [Graphite](https://app.graphite.dev/github/pr/inscreen/html-to-image/22).

**Pull request merged**